client/lib/analytics/ad-tracking/record-order.js

```diff
@@ -432,6 +437,62 @@ function recordOrderInGoogleAds( cart, orderId ) {
 	}
 }
 
+function recordOrderInGAEnhancedEcommerce( cart, orderId, wpcomJetpackCartInfo ) {
+	if ( ! isAdTrackingAllowed() ) {
+		debug( 'recordOrderInGAEnhancedEcommerce: [Skipping] ad tracking is disallowed' );
+		return;
+	}
+
+	if ( ! isGoogleAnalyticsEnabled || ! isGoogleAnalyticsEnhancedEcommerceEnabled ) {
+		debug( 'recordOrderInGAEnhancedEcommerce: [Skipping] Google Analytics is not enabled' );
+		return;
+	}
+
+	let products, brand, totalCostUSD;
+
+	if ( wpcomJetpackCartInfo.containsWpcomProducts ) {
+		products = wpcomJetpackCartInfo.wpcomProducts;
+		brand = GA_PRODUCT_BRAND_WPCOM;
+		totalCostUSD = wpcomJetpackCartInfo.wpcomCostUSD;
+	} else if ( wpcomJetpackCartInfo.containsJetpackProducts ) {
+		products = wpcomJetpackCartInfo.jetpackProducts;
+		brand = GA_PRODUCT_BRAND_JETPACK;
+		totalCostUSD = wpcomJetpackCartInfo.jetpackCostUSD;
+	} else {
+		debug( 'recordOrderInGAEnhancedEcommerce: [Skipping] No products' );
+		return;
+	}
+
+	const items = [];
+	products.map( product => {
+		items.push( {
+			id: product.product_id.toString(),
+			name: product.product_name_en.toString(),
+			quantity: parseInt( product.volume ),
+			price: costToUSD( product.cost, cart.currency ).toString(),
+			brand,
+		} );
+	} );
+
+	const params = [
+		'event',
+		'purchase',
+		{
+			transaction_id: orderId.toString(),
+			value: parseFloat( totalCostUSD ),
+			currency: 'USD',
+			tax: parseFloat( cart.total_tax ) ?? 0,
+			coupon: cart.coupon_code.toString() ?? '',
```

Comment:
I'm seeing an error when there's no coupon code. `toString()` not available.
https://drop.jas.xyz/nLJ9a2

Reply:
commented [here](https://github.com/Automattic/wp-calypso/pull/40683#discussion_r404461711) .